Restructuring of network ACLS (#21)
ansible-network/network.acls: roles/health_checks/README.md

# Health Checks

> The role enables user to manage the ACL resources independent of platforms and perform ACL health checks.

Reviewer: ```suggestion
The role allows users to perform platform-independent Access Control List (ACL) health checks, ensuring consistent and reliable validation across various networking environments.
```

## Capabilities

**ACL Health Checks** :
- Enables users to perform health checks for ACL configurations and their interface assignments.
- Users can perform the following health checks:
  - `available_acls`: Lists all ACLs configured on the device.
  - `missing_acls`: Identifies ACLs referenced in interfaces but not defined.
  - `unassigned_acls`: Shows ACLs that are defined but not used on any interface.
  - `interface_acl_status`: Shows the status of ACL assignments on interfaces.

> This role enables users to create a runtime brownfield inventory with all the ACLS configurations in terms of host vars. These host vars are ansible facts that have been gathered through the *acl_interfaces and *acls network resource module. The tasks offered by this role can be observed below:

Reviewer: I believe it needs to be corrected w.r.t health-checks

Author: yes this line tells the functionality related to the persist role not the heath_check one.

## Variables

| Variable Name        | Default Value | Required | Type | Description                                                  |       Example       |
|:---------------------|:-------------:|:--------:|:----:|:-------------------------------------------------------------|:-------------------:|

> | `ansible_network_os` | `""`          | no       | str  | Network OS to be used during detection.                      | `"cisco.ios.ios"`   |

Reviewer: ```suggestion
| `ansible_network_os` | `""`          | no       | str  | Network OS to be used while performing health-checks.                      | `"cisco.ios.ios"`   |
```

- Enables users to perform health checks for ACL configurations and their interface assignments.
- Users can perform the following health checks:
  - `available_acls`: Lists all ACLs configured on the device.
  - `missing_acls`: Identifies ACLs referenced in interfaces but not defined.
  - `unassigned_acls`: Shows ACLs that are defined but not used on any interface.
  - `interface_acl_status`: Shows the status of ACL assignments on interfaces.

This role enables users to create a runtime brownfield inventory with all the ACLS configurations in terms of host vars. These host vars are ansible facts that have been gathered through the *acl_interfaces and *acls network resource module. The tasks offered by this role can be observed below:

## Variables

| Variable Name        | Default Value | Required | Type | Description                                                  |       Example       |
|:---------------------|:-------------:|:--------:|:----:|:-------------------------------------------------------------|:-------------------:|
| `ansible_network_os` | `""`          | no       | str  | Network OS to be used during detection.                      | `"cisco.ios.ios"` |

> | `data_store`         | `""`          | yes      | dict | Specifies the data store to be used (local or SCM).          | See examples below. |

Reviewer: I believe we don;t need datastore for health checks
